src/near/methods/index.ts

```diff
@@ -42,4 +42,9 @@ export function getMethod(m?: string | null): Schema | undefined {
   if (!m) return undefined
   if (!hasContractMethod(m as MethodName)) return undefined
   return methods[m as MethodName]
+}
+
+export function methodType(m: MethodName): "change" | "view" {
+  if (changeMethods.includes(m)) { return "change" }
+  return "view";
```

Comment:
This array traversal is probably more computationally expensive than the object lookup equivalent:

```suggestion
  const def = topLevelSchema.definitions[m]
  const hasField = hasContractMethodProperty(def)
  return def.contractMethod
```

That has a bunch of stuff to appease TS, but if you wanted to be more JS about it (and if TS imported JSON `as const`, this would probably typecheck):

```suggestion
  return topLevelSchema.definitions[m].contractMethod
```

Reply:
I went with the latter since we have already filtered contract methods at this point.

Reply:
Ok as long as TS is happy with it

Reply:
Not all MethodName keys are legitimate methods; some don't have contractMethod prop